src/annotator/components/Toolbar.tsx

```diff
@@ -1,84 +1,107 @@
  *
  * This is useful to inform assistive technology users when these states
  * have been changed (eg. whether highlights are visible), given that they can
  * be changed in multiple ways (keyboard shortcuts, toolbar button) etc.
- *
- * @param {StatusNotifierProps} props
  */
-function StatusNotifier({ highlightsVisible }) {
+function StatusNotifier({ highlightsVisible }: { highlightsVisible: boolean }) {
   return (
     <div className="sr-only" role="status" data-testid="toolbar-status">
       {highlightsVisible ? 'Highlights visible' : 'Highlights hidden'}
     </div>
   );
 }
 
-/**
- * @typedef ToolbarProps
- *
- * @prop {() => void} closeSidebar -
- *   Callback for the "Close sidebar" button. This button is only shown when
- *   `useMinimalControls` is true and the sidebar is open.
- * @prop {() => void} createAnnotation -
- *   Callback for the "Create annotation" / "Create page note" button. The type
- *   of annotation depends on whether there is a text selection and is decided
- *   by the caller.
- * @prop {boolean} isSidebarOpen - Is the sidebar currently visible?
- * @prop {'annotation'|'note'} newAnnotationType -
- *   Icon to show on the "Create annotation" button indicating what kind of annotation
- *   will be created.
- * @prop {boolean} showHighlights - Are highlights currently visible in the document?
- * @prop {() => void} toggleHighlights -
- *   Callback to toggle visibility of highlights in the document.
- * @prop {() => void} toggleSidebar -
- *   Callback to toggle the visibility of the sidebar.
- * @prop {import("preact").Ref<HTMLButtonElement>} [toggleSidebarRef] -
- *   Ref that gets set to the toolbar button for toggling the sidebar.
- *   This is exposed to enable the drag-to-resize functionality of this
- *   button.
- * @prop {boolean} [useMinimalControls] -
- *   If true, all controls are hidden except for the "Close sidebar" button
- *   when the sidebar is open. This is enabled in the "clean" theme.
- */
+export type ToolbarProps = {
+  /**
+   * Callback for the "Close sidebar" button. This button is only shown when
+   * `useMinimalControls` is true and the sidebar is open.
+   */
+  closeSidebar: () => void;
+
+  /** Callback for when "Create annotation" button is clicked. */
+  createAnnotation: () => void;
+
+  /** Is the sidebar currently open? */
+  isSidebarOpen: boolean;
+
+  /**
+   * Informs which icon to show on the "Create annotation" button and what type
+   * of annotation should be created by the `createAnnotation` callback. The
+   * type of annotation depends on whether there is a text selection in the
+   * document.
+   */
+  newAnnotationType: 'annotation' | 'note';
+
+  /** Are highlights currently visible in the document? */
+  showHighlights: boolean;
+
+  /** Callback for the show/hide highlights button */
+  toggleHighlights: () => void;
+
+  /** Callback for toggling the visibility of the sidebar when the show/hide
+   * sidebar button is clicked */
```

Comment:
Since this is a multi-line comment, the `/**` and trailing `*/` should appear on their own lines.